src/checks/hyper-training-31.js

```diff
@@ -0,0 +1,21 @@
+module.exports = {
+  description: 'A maxed IV (31) cannot be hypertrained.',
+  filter (pkmn) {
+    return pkmn.ivHp === 31 || pkmn.ivAtk === 31 || pkmn.ivDef === 31 ||
+        pkmn.ivSpAtk === 31 || pkmn.ivSpDef === 31 || pkmn.ivSpe === 31;
+  },
+  field: 'iv',
+  test (pkmn) {
+    const maxIV = {
+      HP: pkmn.ivHp === 31,
+      Atk: pkmn.ivAtk === 31,
+      Def: pkmn.ivDef === 31,
+      SpAtk: pkmn.ivSpAtk === 31,
+      SpDef: pkmn.ivSpDef === 31,
+      Spe: pkmn.ivSpe === 31
+    };
+    return !(maxIV.Hp ? pkmn.hyperTrainedHp : maxIV.Hp || maxIV.Atk ? pkmn.hyperTrainedAtk : maxIV.Atk ||
```

Comment:
I think these ternary expressions are mostly redundant -- if `maxIV.Hp` is false, then there's no reason to negate it again. This check could be simplified to

```js
return !(
  pkmn.ivHp === 31 && pkmn.hyperTrainedHp ||
  pkmn.ivAtk === 31 && pkmn.hyperTrainedAtk ||
  pkmn.ivDef === 31 && pkmn.hyperTrainedDef ||
  pkmn.ivSpAtk === 31 && pkmn.hyperTrainedSpAtk ||
  pkmn.ivSpDef === 31 && pkmn.hyperTrainedSpDef ||
  pkmn.ivSpe === 31 && pkmn.hyperTrainedSpe
);
```